skimage/feature/texture.py

```diff
@@ -291,3 +293,130 @@ def local_binary_pattern(image, P, R, method='default'):
     return output
+
+def multiblock_local_binary_pattern(int_image, x, y, width, height):
+    """Multi-block local binary pattern.
+
+    MB-LBP is an extension of LBP that can be computed on many
+    scales in a constant time using integral image. It consists of
+    9 equal-sized rectangles. Sum of pixels' intensity values
+    in each of them are compared to the central rectangle and
+    depending on comparison result, the feature descriptor is
+    computed.
+
+    Parameters
+    ----------
+    int_image : (N, M) array
+        Integral image.
+    x : int
+        X-coordinate of top left corner of a rectangle containing feature.
+    y : int
+        Y-coordinate of top left corner of a rectangle containing feature.
+    width : int
+        Width of one of 9 equal rectangles that will be used to compute
+        a feature.
+    height : int
+        Height of one of 9 equal rectangles that will be used to compute
+        a feature.
+
+    Returns
+    -------
+    output : int
+        8bit MB-LBP feature descriptor.
+
+    References
+    ----------
+    .. [1] Face Detection Based on Multi-Block LBP
+           Representation. Lun Zhang, Rufeng Chu, Shiming Xiang, Shengcai Liao,
+           Stan Z. Li
+           http://www.cbsr.ia.ac.cn/users/scliao/papers/Zhang-ICB07-MBLBP.pdf
+    """
+
+    # Top-left coordinates of central rectangle
+    central_rect_x = x + width
+    central_rect_y = y + height
+
+    # Sum of intensity values of central rectangle
+    central_rect_val = integrate(int_image,
+                                 central_rect_y,
+                                 central_rect_x,
+                                 central_rect_y + height - 1,
+                                 central_rect_x + width - 1)
+
+    # Offsets of neighbour rectangles relative to central one.
+    # It has order starting from top left and going clockwise
+    neighbour_rect_offsets = ((-1, -1), (0, -1), (1, -1),
+                              (1, 0), (1, 1), (0, 1),
+                              (-1, 1), (-1, 0))
+
+    lbp_code = 0
+
+    for element_num, offset in enumerate(neighbour_rect_offsets):
+
+        offset_x, offset_y = offset
+
+        current_rect_x = central_rect_x + offset_x * width
+        current_rect_y = central_rect_y + offset_y * height
+        current_rect_val = integrate(int_image,
+                                     current_rect_y,
+                                     current_rect_x,
+                                     current_rect_y + height - 1,
+                                     current_rect_x + width - 1)
+
+        has_greater_value = current_rect_val >= central_rect_val
+
+        # If current rectangle's intensity value is bigger
+        # make corresponding bit to 1.
+        lbp_code |= has_greater_value << element_num
+
+        print lbp_code
+
+    return lbp_code
+
+def visualize_multiblock_lbp(img, x, y, width, height, lbp_code=0):
```

Comment:
I think for consistency we can make this resemble the [rag_draw](https://github.com/scikit-image/scikit-image/blob/master/skimage/future/graph/rag.py#L325) function. Rather than plotting here, we can return a new image.

@stefanv , `rag_draw` import `matplotlib` at the start, is that a problem ? Because of at all, it will only add up to the start up time of the program.